File: server/controller/accessimpl.cpp
#include "../common.hpp"

#include "accessimpl.hpp"

::grpc::Status AccessImpl::Echo(::grpc::ServerContext* context,
                                const ::stq::Empty* request,
                                ::stq::EchoRes* response)
{
    UNUSED(context);
    UNUSED(request);
    UNUSED(response);
    return ::grpc::Status::OK;
}

File: server/controller/grpc/accessimpl.go
package grpc

import (
	pb "STQ/protos"
	"context"
)

type accessImpl struct {
	pb.UnimplementedAccessServer
}

func (a *accessImpl) Echo(_ context.Context, _ *pb.Empty) (*pb.EchoRes, error) {
	return &pb.EchoRes{}, nil
}

func (a *accessImpl) Stop(_ context.Context, _ *pb.Empty) (*pb.EchoRes, error) {
	global.Logger.Info().Msg("Received stop signal from grpc. stopping...")
	global.StopChan <- true
	return &pb.EchoRes{}, nil
}
